Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1356,6 +1356,15 @@ LayoutUnit RenderBlockFlow::collapseMarginsWithChildInfo(RenderBox* child, Rende
     return logicalTop;
 }
 
+bool RenderBlockFlow::shouldTrimChildMargin(MarginTrimType marginTrimType, const RenderBox& child) const
+{
+    if (!child.style().isDisplayBlockLevel() || !style().marginTrim().contains(marginTrimType))
+        return false;
+    if (marginTrimType == MarginTrimType::BlockStart)
+        return firstInFlowChildBox() == &child;
+    return lastInFlowChildBox() == &child;
+}
+
```

Comment:
how does this work with non-adjoining margins? -we don't even end up here?

Reply:
I don't have any test cases with non-adjoining margins (all of the container margins are 0). For the following example: 

`<style>
.container {
    margin-block-start: 10px;
    padding-block-start: 10px;
    background-color: green;
    margin-trim: block-start;
}
.child {
    width: 100%;
    height: 100px;
    margin-block-start: 20px;
}
</style>

<container>
<child></child>
</container>`

Do you think that the block-start margin of the child should not be trimmed since it is non-adjoining? I feel like it should be trimmed and set to 0px since the child's margin is still adjacent to the container's block-start edge (unless it means the margin edge).

I have another bug for collapsed margins so I was planning on dealing with them in another patch.

Reply:
@sgill26 Can you adjust the title a bit to reflect that there are more steps to come related to collapsed margins, etc. Maybe call this "(Step 1)" or "(Initial Implementation)" or something to clarify to people looking at changelogs in the future that this isn't the entire set of changes needed.

Reply:
Done